python/cudf/cudf/io/json.py

```diff
@@ -111,25 +91,23 @@ def read_json(
             "be GPU accelerated in the future"
         )
 
-        if not ioutils.ensure_single_filepath_or_buffer(
-            path_or_data=path_or_buf,
-            storage_options=storage_options,
-        ):
-            raise NotImplementedError(
-                "`read_json` does not yet support reading "
-                "multiple files via pandas"
+        filepaths_or_buffers, compression = (
+            ioutils.get_reader_filepath_or_buffer(
+                path_or_data=path_or_buf,
+                compression=compression,
+                iotypes=(BytesIO, StringIO),
+                allow_raw_text_input=True,
+                storage_options=storage_options,
             )
-
-        path_or_buf, compression = ioutils.get_reader_filepath_or_buffer(
-            path_or_data=path_or_buf,
-            compression=compression,
-            iotypes=(BytesIO, StringIO),
-            allow_raw_text_input=True,
-            storage_options=storage_options,
         )
+        if len(filepaths_or_buffers) > 1:
```

Comment:
I think I'd like to avoid adding more stuff to `get_reader_filepath_or_buffer` (if possible), but I agree that the repeated logic can be consolidated in one place. I added a `_select_single_source` utility for this. Let me know if you think this makes sense or if another `get_reader_filepath_or_buffer` would be best.

Reply:
Works for me.